src/EditorFeatures/CSharp/BraceMatching/DirectiveTriviaMatcher.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.BraceMatching
+{
+    [ExportBraceMatcher(LanguageNames.CSharp)]
+    internal class DirectiveTriviaMatcher : IBraceMatcher
+    {
+        public async Task<BraceMatchingResult?> FindBracesAsync(Document document, int position, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var token = root.FindToken(position, findInsideTrivia: true);
+
+            var directive = token.Parent as DirectiveTriviaSyntax;
```

Comment:
It works only when the caret is on the hashtoken or the keyword (`#if$$` `$$#else`) and not on identifiers (`$$CHK`) or related trailing trivia (`$$MyRegion`). 
In that it is consistent with how Highlight references performs navigation on these and with brace matching itself in VS2013. I'm open to tweaking this if you feel we should.